File: github.php
<?php
require_once('workflows.php');

class Github {
	public $repo;
	private $token;
	private $w;
	
	function __construct($repo, $token=null) {
		$this->repo = $repo;
		$this->token = $token;
		$this->w = new Workflows();
	}

	public function getContent() {
		return $this->query("repos/".$this->repo."/contents");
	}

	public function search($query) {
		return $this->query("search/code", array("q" => "$query repo:".$this->repo));
	}

	private function query($resource, $params = null) {
		$url = $this->getUrl($resource, $params);
		$jsonResponse = $this->w->request($url, array(CURLOPT_USERAGENT => 'alfred'));
		$response = json_decode($jsonResponse, true);
		return $response;
	}
	
	private function getUrl($resource, $params = null) {
		$allParams = is_null($params) ? array() : $params;
		if(!empty($this->token)) {
			$allParams["access_token"] = $this->token;
		}
		
		$url = sprintf("%s?%s", "https://api.github.com/".$resource, http_build_query($allParams));
		return $url;
	}
}

?>

File: src/notebook/github.py
from workflow import web

class GitHub:
    def __init__(self, repo, token = None):
        self.repo = repo
        self.token = token

    def search(self, query):
        url = "https://api.github.com/search/code"
        fullQuery = query + " in:file,path"
        params= {'q': fullQuery + ' repo:' + self.repo}
        return self.getItems(url, params)['items']

    def ls(self, path = '/'):
        url = "https://api.github.com/repos/" + self.repo + "/contents" + path
        return self.getItems(url)

    def getUrlOfItem(self, item):
        return "https://github.com/" + self.repo + "/blob/master/" + item['path'].replace (" ", "%20")

    def getItems(self, url, params = {}):
        if self.token is not None:
            params['access_token'] = self.token
        return web.get(url, params).json()